### PR TITLE
Fixes #2748. Pull GA params from POST data and send them inline

### LINKED ISSUE #2748: Forward GA params to new issue page
As of https://bugzilla.mozilla.org/show_bug.cgi?id=1514861, we'll be sending utm_source and utm_campaign params from the browser, and when https://github.com/webcompat/webcompat-reporter-extensions/pull/141 lands, from addons.

From the browser, we POST these params, but we'll need to send them along when we render the bug form (I think).

### PR DESCRIPTION
**Pull request by miketaylr**
This PR gives us the ability to use CSP nonce attributes in inline JS, and parses out the GA params from https://bugzilla.mozilla.org/show_bug.cgi?id=1514861.

Thinking about how on earth to test this...

### CODE REVIEW

File: webcompat/helpers.py [hunk 1]
```diff
@@ -508,7 +508,7 @@ def add_csp(response):
         "font-src 'self' https://fonts.gstatic.com; " +
         get_img_src_policy() +
         "manifest-src 'self'; " +
-        "script-src 'self' https://www.google-analytics.com https://api.github.com; " +  # nopep8
+        "script-src 'self' https://www.google-analytics.com https://api.github.com 'nonce-{nonce}';".format(nonce=request.nonce) +  # nopep8
```
Comment (miketaylr) [inline webcompat/helpers.py]: nit: a space after `;` would be good.

File: webcompat/templates/new-issue.html [hunk 2]
```diff
@@ -8,3 +8,13 @@
   </section>
 </main>
 {% endblock %}
+{%- block extrascripts -%}
+{% if source and campaign %}
+<script nonce={{nonce}}>
+if (typeof ga == "function") {
```
Comment (miketaylr) [inline webcompat/templates/new-issue.html]: This typecheck is just to prevent typeerrors locally (we don't load GA).

File: webcompat/views.py [hunk 3]
```diff
@@ -55,6 +57,7 @@ def before_request():
         g.user = User.query.get(session['user_id'])
     g.referer = get_referer(request) or url_for('index')
     g.request_headers = request.headers
+    request.nonce = sha1(uuid4().hex).hexdigest()
```
Comment (miketaylr) [inline webcompat/views.py]: We can move to https://docs.python.org/3/library/secrets.html one day...

**Comment by miketaylr** [comment]
@marimeireles can you review this please? thanks!

File: tests/unit/test_urls.py [hunk 4]
```diff
@@ -256,5 +257,76 @@ def test_webhooks_route(self):
...
+            'user_agent': u'BurgerJSON',
+        }
+        rv = self.app.post(
+            '/issues/new?url=http://example.net/&src=web&label=type-stylo',
+            headers=headers, json=json_data)
+        self.assertEqual(rv.status_code, 200)
+        # do we not have a <script nonce=hash> in our response body?
+        self.assertNotIn('<script nonce=', rv.data)
+
+
```
Comment (marimeireles) [inline tests/unit/test_urls.py]: The tests seems pretty straightforward. Thanks for the comments, they helped a lot :)

**Comment by miketaylr** [comment]
Seems like a terrible idea to deploy this on a Friday before winter PTO, while @karlcow is also on PTO. So let's wait (and it will give @karlcow a chance to take a look as well).

**Comment by karlcow** [comment]
> Seems like a terrible idea to deploy this on a Friday before winter PTO, while @karlcow is also on PTO. So let's wait (and it will give @karlcow a chance to take a look as well).

terrible ideas are fun. Back from 🌴 🐘

File: tests/unit/test_urls.py [hunk 5]
```diff
@@ -256,5 +257,76 @@ def test_webhooks_route(self):
...
+    def test_extracted_ga_params_end_up_as_inline_js(self):
+        """Extract GA params (utm_foo) information of a form request with a POST.
+
+        We also test that the nonce in the CSP matches the nonce in the
+        inline style.
+        """
+        nonce_re = r'(?:nonce\-)(?P<nonce>[a-z0-9]+)'
+        json_data = {
+            'user_agent': u'BurgerJSON',
+            'utm_source': u'mcdonalds',
```
Comment (karlcow) [inline tests/unit/test_urls.py]: poor Illegal Burger…

**Comment by karlcow** [comment]
We can probably deploy. :)

**Pull request merged**